### DISCUSSION TITLE
KubernetesPodOperator launch Deployment or ReplicaSet instead of Pod

### DISCUSSION THREAD

**Discussion by veerendra2** [apache/airflow#28887] [Q&A] [answered]
Hello,
I'm very new to apache airflow. While checking docs, I see the `KubernetesPodOperator` only launches Pods for DAG context. Is it possible run Deployment or ReplicaSet instead of Pod

Our K8s cluster are configured with Kyverno policies to NOT run standaload pods. 

Any workaround would be also helpful 

Thanks

**Comment by potiuk** [comment, marked as answer]
This makes no sense to run a task to be a Deployment or ReplicaSet. Hence `Pod` in the name. But you do not have to execute KPO to run tasks with Airflow- the KPO is only needed when you want isolated Pod running separately from Airflow task - but you can easily run any tasks  - including many specialized operators as well as generic BashOperator, PythonOperator.

If you need isolation you can also use DockerOperator - for that one you need to just have a docker engine running and ability to start containers. Or you can have another, separate Kubernetes Cluster which would be able to connect to and continue using KPO - KPO has the possibilty to connnect to another cluster.

Also if you need another degree of isolation there are PythonVirtualenvOperator and recently ExternalPythonOperator that can run Python code in a separate virtualenv. 

All the options you have for the isolation are nicely summarized here: https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#handling-conflicting-complex-python-dependencies
